api: implement a `next` page after authentification

privat opened the pull request:
TODO: The UI part (www/javascripts/ng-app.js & www/directives/player/menu.html)

Morriar commented on src/api/api_auth_shibuqam.nit:
```diff
@@ -94,7 +99,10 @@ class ShibCallback
-		res.redirect "/player"
+
+		# Redirect
+		var page = session.auth_next or else "/player"
```
Maybe this should be pulled up in the common ancestor of Github and Shibuqam login?

privat replied:
What about adding common services in `APIHandler` ?

Morriar replied:
The next page one? Maybe, do you have another example than login/logout where the `next` information is useful?

I can see the mission one but I don't like the semantics of passing the next uri in one case and guessing it in another case.

Following the current popcorn philosophy, this should be in the AuthHandler.

privat replied:
AuthHadler is not documented, I not sure what is its role but it seems unrelated to the classes GithubLogin, GithubCallback, ShibLogin and ShibCallback.

Morriar replied:
My bad, I though it was this one. It appears that ShibUqam and Github does not have a common ancestor more specific than APIHandler...

But I don't like the id of putting everything in APIHandler, it makes it API dirtier :/

privat replied:
If I introduce a common abstract superclass in api_auth with the common services/code, is that OK for you?

privat replied:
Eg `abstract class AuthLogin`

Morriar replied:
Perfect

privat committed:
api_auth: introduce `AuthLogin` to factorise common services among github&shib

Signed-off-by: Jean Privat <jean@pryen.org>

privat committed:
api: implement a `next` page after authentification

TODO: The UI part (www/javascripts/ng-app.js & www/directives/player/menu.html)

privat commented:
reroll with a prior factorization of services common to github and shib

ppepos commented:
Noice! +1

lbajolet commented on src/api/api_auth.nit:
```diff
@@ -57,6 +60,48 @@ class Logout
+abstract class AuthLogin
+	super APIHandler
+
+	# Extract a possible next page from the GET arguments and store it in the session for latter
```
:s/latter/later

lbajolet commented:
+1
